GiniVision/Classes/Core/Screens/Multipage Review/MultipageReviewCollectionsCellPresenter.swift

```diff
@@ -0,0 +1,209 @@
+//
+//  MultipageReviewCollectionCellPresenter.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 12/28/18.
+//
+
+import Foundation
+
+protocol MultipageReviewCollectionCellPresenterDelegate: class {
+    func multipage(_ reviewCollectionCellPresenter: MultipageReviewCollectionCellPresenter,
+                   didUpdate cell: MultipageReviewCollectionCellPresenter.MultipageCollectionCellType,
+                   at indexPath: IndexPath)
+}
+
+final class MultipageReviewCollectionCellPresenter {
+    
+    weak var delegate: MultipageReviewCollectionCellPresenterDelegate?
+    var thumbnails: [String: [ThumbnailType: UIImage]] = [:]
+    fileprivate let giniConfiguration: GiniConfiguration
+    fileprivate let thumbnailsQueue = DispatchQueue(label: "Thumbnails queue")
+    
+    enum MultipageCollectionCellType {
+        case main(MultipageReviewMainCollectionCell, (NoticeActionType) -> Void)
+        case pages(MultipageReviewPagesCollectionCell)
+    }
+    
+    enum ThumbnailType {
+        case big, small
+        
+        var scale: CGFloat {
+            switch self {
+            case .big:
+                return 1.0
+            case .small:
+                return 1/4
```

Comment:
My intuition is that 25% is still too much for a thumbnail. But maybe it gets distorted. Did you try lower values?

Reply:
Yes, I tried both 1/5 and 1/6 and it got distorted. Plus the performance with 1/4 is quite good.